build.gradle

```diff
@@ -112,7 +113,8 @@ dependencies {
     testCompile(javadocJDKFiles)
 
     compile 'org.broadinstitute:barclay:1.2.1'
-    compile 'com.github.broadinstitute:picard:2.12.2'
+    compile 'com.github.broadinstitute:picard:' + picardVersion
+    compileOnly 'com.github.broadinstitute:picard:' + picardVersion + ':sources'
```

Comment:
I'd add a new configuration for this, to being able to include more in the future:

```gradle
configurations {
    ## requires the ":sources" marker
    docgen {
        transitive false
    }
}

..

dependencies {
    ..

    docgen 'com.github.broadinstitute:picard:' + picardVersion + ':sources'

    ..
}
```